// mydb/index.js
var Database = require('better-sqlite3');
var db = new Database('public/dat/db/dat1.db');
var sku;

exports.selTyp=function(type){
var stm=db.prepare('select * from axe where type=?');

try{
var row=stm.all(type);
//sku=row.sku;
    
}catch(err){console.log(err);}

return row;

};
exports.selTmp=function(arg2){
var stm=db.prepare('select * from tmp where usr=?');

try{
var row=stm.all(arg2);
//sku=row.sku;
    
}catch(err){console.log(err);}

return row;

};

exports.selMer=function(){
var stm=db.prepare('select * from mer');

try{
var row=stm.all();
}catch(err){console.log(err);}
return row;

};

exports.updTmp=function(uni,usr,sku){
var stm=db.prepare('update tmp set uni=? where usr=? and sku=? ');

try{
stm.run(uni,usr,sku);
}catch(err){console.log(err);}
};

exports.insTmp=function(usr,sku,uni){
var stm=db.prepare('insert into tmp values(?,?,?) ');

try{
stm.run(usr,sku,uni);
    
}catch(err){console.log(err);}
};


exports.cntMer=function(){
var stm=db.prepare('select sku from mer');

try{
var row=stm.all();
}catch(err){console.log(err);}
return row.length;

};
