// aidb/index.js
var Database = require('better-sqlite3');
var db = new Database('public/dat/db/aid.db');

// === usr ===
exports.selUsr=function(arg2){
var stm=db.prepare('select * from usr where usr=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};
exports.allUsr=function(){
var stm=db.prepare('select * from usr ');
var row=stm.all();
return row;
};
exports.phnUsr=function(arg2){
var stm=db.prepare('select * from usr where phn=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

// === adr ===
exports.phnAdr=function(phn){
var stm=db.prepare('select * from adr where phn=? ');
var row=stm.get(phn);
return row;
};
// === usr run ===
exports.updusr=function(uni,usr,sku){
var stm=db.prepare('update usr set uni=? where usr=? and sku=? ');
try{stm.run(uni,usr,sku);
}catch(err){console.log(err);}
};

exports.insusr=function(usr,sku,uni){
var stm=db.prepare('insert into usr values(?,?,?) ');
try{stm.run(usr,sku,uni);
}catch(err){console.log(err);}
};

exports.delusr=function(){
var stm=db.prepare('delete from usr where uni=0');
try{stm.run();
}catch(err){console.log(err);}
};
// === mer ===
exports.allMer=function(){
var stm=db.prepare('select *,rowid from mer');
try{var row=stm.all();
}catch(err){console.log(err);}
return row;
};

exports.skuMer=function(sku){
var stm=db.prepare('select * from mer where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};

exports.insMer=function(name,sku,pri,img,rel,cat,des){
var stm=db.prepare('insert into mer values(?,?,?,?,?,?,?)');
stm.run(name,sku,pri,img,rel,cat,des);
};

// === aid ===
exports.allAid=function(){
var stm=db.prepare('select * from aid');
try{var row=stm.all();}catch(err){console.log(err);}
return row;
};
exports.insAid=function(usr,pss,email,phn){
var stm=db.prepare('insert into aid values(?,?,?,?)');
stm.run(usr,pss,email,phn);
};
exports.updAid=function(pid,usr){
var stm=db.prepare('update aid set pid=? where usr=?');
stm.run(pid,usr);
};
exports.ordAid=function(usr,pid,name,mnt,dat,stat){
var stm=db.prepare('insert into aid_ord values(?,?,?,?,?,?)');
stm.run(usr,pid,name,mnt,dat,stat);
};
